src/libcore/iter.rs

```diff
@@ -2655,6 +2656,7 @@ pub fn range_step<A: Int>(start: A, stop: A, step: A) -> RangeStep<A> {
 }
 
 #[stable(feature = "rust1", since = "1.0.0")]
+#[allow(deprecated)]
```

Comment:
This seems somewhat worrisome to use these deprecated bounds in stable implementations. I think we had a problem in the past, for example, about an impl of `Iterator` for each concrete primitive type as a type parameter causes type inference to go awry. I forget the details, but we may want to deal with this sort of fallout sooner rather than later.